pkg/controller/upload_controller.go

```diff
@@ -189,10 +195,20 @@ func (c *uploadController) pvbHandler(obj interface{}) {
 	log := loggerForUpload(c.logger, req)
 
 	switch req.Status.Phase {
-	case "", pluginv1api.UploadPhaseNew, pluginv1api.UploadPhaseInProgress:
+	case "", pluginv1api.UploadPhaseNew, pluginv1api.UploadPhaseInProgress, pluginv1api.UploadPhaseUploadError:
 		// Process New and InProgress Uploads
 	default:
-		log.Debug("Upload CR is not New or InProgress, skipping")
+		log.Debugf("Upload CR is not New or InProgress or UploadPhaseUploadFailed, skipping")
+		return
+	}
+
+	log.Infof("Filtering out the retry upload request which comes in before next retry time")
+	if req.Status.RetryCount > MIN_RETRY && c.clock.Now().Unix() < req.Status.NextRetryTimestamp.Unix() {
```

Comment:
use a single variable for the c.clock.Now().Unix() to make the one in if check and the one in log message in sync.